lib/good_job/job.rb

```diff
@@ -126,6 +126,8 @@ def self.queue_parser(string)
     # @return [ActiveRecord::Relation]
     scope :display_all, (lambda do |after_scheduled_at: nil, after_id: nil|
       query = order(Arel.sql('COALESCE(scheduled_at, created_at) DESC, id DESC'))
+      after_scheduled_at = Time.zone.parse(after_scheduled_at) if after_scheduled_at.present?
```

Comment:
This will fail if `after_scheduled_at` is a `Time` or `DateTime`. My preference would be that type coercion happens in the controller (close to where it's coming in as a string), and models operate off of higher level objects (like `Time`/`DateTime`) rather than primitives (like Strings)

Reply:
updated